cmd2.py

```diff
@@ -193,7 +193,7 @@ def flag_based_complete(text, line, begidx, endidx, flag_dict, default_completer
         return []
 
     completions = []
-    flag_processed = False
+    flag_present = False
```

Comment:
As far as I can tell, these changes look good.  You might consider modifying either the tab_completion.py or subcommands.py example to illustrate the case where this PR would matter.